tests/signup.spec.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import toJson from 'enzyme-to-json';
+import renderer from 'react-test-renderer';
+import configureStore from 'redux-mock-store';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
+import { format } from 'util';
+import SignupPage from '../src/pages/Signup/index';
+import { shallow, mount, render } from './enzyme';
+import Schema from '../src/pages/Signup/schema';
+
+const mockStore = configureStore();
+const store = mockStore({
+  signupReducer: {
+    isLoading: false,
+    error: {},
+    isCompleted: false,
+    signUp: jest.fn()
+  }
+});
+const fake = jest.fn();
+
+describe('it should work', () => {
+  let wrapper = null;
+  beforeEach(() => {
+    wrapper = mount(
+      <Provider store={store}>
+        <Router>
+          <SignupPage signUp={fake} />
+        </Router>
+      </Provider>
+    );
+  });
+
+  it('If it works', () => {
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('renders the component', () => {
+    expect(wrapper.exists()).toBe(true);
+    expect(wrapper.find('Form').exists()).toBe(true);
+    expect(wrapper.find('button').exists()).toBe(true);
```

Comment:
I think asserting that there's a submit button (type=submit)/ the text in the button is more useful than just that there's a button.

Assert the number of buttons. Other important form elements as well.

Reply:
This test doesn't actually test that the component signs a user up. And that is reasonable because the component doesn't sign a user up. The action/network request does that. The component only displays what is appropriate based on the application state.

Also, add test for negative paths e.g when a user enters invalid data.

**Hypothetical Solution**
```js
const  jest.fn(() => 'lol')

state =  { signUp }

// then in your test.

sbmitBtn.simulate('click');

// then you should be able to do:

expect(signUp).toHaveBeenCalled();
expect(signUp).toHaveBeenCalledTimes(1);

```